store/searchtest/user_layer.go

```diff
@@ -6,674 +6,891 @@ package searchtest
+	})
+	t.Run("Should autocomplete users that share the same email domain", func(t *testing.T) {
+		users, apperr := th.Store.User().AutocompleteUsersInChannel(th.Team.Id, th.ChannelBasic.Id, "simulator.amazon.com", options)
+		require.Nil(t, apperr)
+		th.User.Sanitize(map[string]bool{})
+		th.User2.Sanitize(map[string]bool{})
+		th.assertUsersMatchInAnyOrder(t, []*model.User{th.User}, users.InChannel)
+		th.assertUsersMatchInAnyOrder(t, []*model.User{th.User2}, users.OutOfChannel)
+	})
+	t.Run("Should search users when the email is unique", func(t *testing.T) {
+		users, apperr := th.Store.User().Search(th.Team.Id, "useralt@test.email.com", options)
+		require.Nil(t, apperr)
+		userAlternate.Sanitize(map[string]bool{})
+		th.assertUsersMatchInAnyOrder(t, []*model.User{userAlternate}, users)
+	})
+	t.Run("Should search users that share the same email user prefix", func(t *testing.T) {
+		users, apperr := th.Store.User().Search(th.Team.Id, "success_", options)
+		require.Nil(t, apperr)
+		th.User.Sanitize(map[string]bool{})
+		th.User2.Sanitize(map[string]bool{})
+		th.assertUsersMatchInAnyOrder(t, []*model.User{th.User}, users)
+	})
+	t.Run("Should search users that share the same email domain", func(t *testing.T) {
+		users, apperr := th.Store.User().Search(th.Team.Id, "simulator.amazon.com", options)
+		require.Nil(t, apperr)
+		th.User.Sanitize(map[string]bool{})
+		th.User2.Sanitize(map[string]bool{})
+		th.assertUsersMatchInAnyOrder(t, []*model.User{th.User}, users)
+	})
+}
+func testShouldNotMatchSpecificQueriesEmail(t *testing.T, th *SearchTestHelper) {
+	options := &model.UserSearchOptions{
+		AllowEmails: false,
+		Limit:       model.USER_SEARCH_DEFAULT_LIMIT,
+	}
+	users, apperr := th.Store.User().AutocompleteUsersInChannel(th.Team.Id, th.ChannelBasic.Id, "success_", options)
+	require.Nil(t, apperr)
+	th.assertUsersMatchInAnyOrder(t, []*model.User{}, users.InChannel)
+	th.assertUsersMatchInAnyOrder(t, []*model.User{}, users.OutOfChannel)
+}
+func testAutocompleteUserByUsernameWithDot(t *testing.T, th *SearchTestHelper) {
+	userAlternate, err := th.createUser("alternate.username", "alternatenickname", "firstname", "altlastname")
 	require.Nil(t, err)
-	_, err = s.Channel().SaveMember(&model.ChannelMember{ChannelId: channel.Id, UserId: u3.Id, NotifyProps: model.GetDefaultChannelNotifyProps()})
+	defer th.deleteUser(userAlternate)
+	err = th.addUserToTeams(userAlternate, []string{th.Team.Id})
 	require.Nil(t, err)
-	_, err = s.Team().SaveMember(&model.TeamMember{TeamId: team.Id, UserId: u5.Id}, -1)
+	_, err = th.addUserToChannels(userAlternate, []string{th.ChannelBasic.Id})
 	require.Nil(t, err)
-	_, err = s.Channel().SaveMember(&model.ChannelMember{ChannelId: channel.Id, UserId: u5.Id, NotifyProps: model.GetDefaultChannelNotifyProps()})
+	options := &model.UserSearchOptions{
+		Limit: model.USER_SEARCH_DEFAULT_LIMIT,
+	}
+	// Should return results when searching for the whole username with Dot
```

Comment:
nit: unnecessary as it's in the name of the test

Reply:
Garbage I forgot to remove :sweat_smile: